api-reference/_hidden/GridBaseColumn/headerFilter/dataSource.md

```diff
@@ -17,4 +17,23 @@ The UI component's instance.
 A DataSource configuration.
 
 ---
-The {WidgetName} generates a header filter's data source automatically based on column values. Use the **dataSource** property to change the generated data source or specify a custom data source. Refer to the following help topic for more information: [Customize Header Filter Data Source](/concepts/05%20UI%20Components/DataGrid/99%20How%20To/Customize%20Header%20Filter%20Data%20Source '/Documentation/Guide/UI_Components/DataGrid/How_To/Customize_Header_Filter_Data_Source/').
+
+This property accepts one of the following:
+
+- **Array of objects or primitives**       
+A JavaScript array that contains plain objects or primitives.
```

Comment:
AFAIK, primitives are not supported. This use case is described in the [Specify a Custom Data Source](https://js.devexpress.com/Documentation/Guide/UI_Components/DataGrid/How_To/Customize_Header_Filter_Data_Source/#Specify_a_Custom_Data_Source) article. Let's direct the reader there.